pkg/operator/apiserver/controller/auditpolicy/auditpolicy_controller.go

```diff
@@ -0,0 +1,136 @@
+	kubeClient kubernetes.Interface,
+	configInformers configinformers.SharedInformerFactory,
+	kubeInformersForTargetNamesace kubeinformers.SharedInformerFactory,
+	eventRecorder events.Recorder,
+) factory.Controller {
+	c := &AuditPolicyController{
+		operatorClient:        operatorClient,
+		apiserverConfigLister: apiserverConfigLister,
+		kubeClient:            kubeClient,
+		targetNamespace:       targetNamespace,
+		targetConfigMapName:   targetConfigMapName,
+	}
+
+	return factory.New().WithSync(c.sync).ResyncEvery(10*time.Second).WithInformers(
+		configInformers.Config().V1().APIServers().Informer(),
+		kubeInformersForTargetNamesace.Core().V1().ConfigMaps().Informer(),
+	).ToController("AuditPolicyController", eventRecorder.WithComponentSuffix("audit-policy-controller"))
+}
+
+func (c *AuditPolicyController) sync(ctx context.Context, syncCtx factory.SyncContext) error {
+	operatorConfigSpec, _, _, err := c.operatorClient.GetOperatorState()
+	if err != nil {
+		return err
+	}
+
+	switch operatorConfigSpec.ManagementState {
+	case operatorsv1.Managed:
+	case operatorsv1.Unmanaged:
+		return nil
+	case operatorsv1.Removed:
+		errs := []error{}
+		if err != nil {
+			errs = append(errs, err)
+			return errors.NewAggregate(errs)
+		}
+		if err := c.kubeClient.CoreV1().ConfigMaps(c.targetNamespace).Delete(ctx, c.targetConfigMapName, metav1.DeleteOptions{}); err != nil {
+			errs = append(errs, err)
+		}
+		return errors.NewAggregate(errs)
+	default:
+		syncCtx.Recorder().Warningf("ManagementStateUnknown", "Unrecognized operator management state %q", operatorConfigSpec.ManagementState)
+		return nil
+	}
+
+	config, err := c.apiserverConfigLister.Get("cluster")
+	if err != nil {
+		return err
+	}
+
+	err = c.syncAuditPolicy(ctx, config.Spec.Audit, syncCtx.Recorder())
+
+	// update failing condition
+	cond := operatorv1.OperatorCondition{
+		Type:   "AuditPolicyDegraded",
+		Status: operatorv1.ConditionFalse,
+	}
+	if err != nil {
+		cond.Status = operatorv1.ConditionTrue
+		cond.Reason = "Error"
+		cond.Message = err.Error()
+	}
+	if _, _, updateError := v1helpers.UpdateStatus(c.operatorClient, v1helpers.UpdateConditionFn(cond)); updateError != nil {
+		if err == nil {
+			return updateError
+		}
+	}
+
+	return err
+}
+
+func (c *AuditPolicyController) syncAuditPolicy(ctx context.Context, config configv1.Audit, recorder events.Recorder) error {
+	desired, err := audit.GetAuditPolicy(config)
+	if err != nil {
+		return err
+	}
+	desired = desired.DeepCopy()
+	desired.Kind = "Policy"
+	desired.APIVersion = auditv1.SchemeGroupVersion.String()
+
+	bs, err := yaml.Marshal(desired)
+	if err != nil {
+		return err
+	}
+
+	cm := &v1.ConfigMap{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: c.targetNamespace,
+			Name:      c.targetConfigMapName,
+		},
+		Data: map[string]string{
+			"policy.yaml": string(bs),
+		},
+	}
+
+	_, _, err = resourceapply.ApplyConfigMap(c.kubeClient.CoreV1(), recorder, cm)
+	return err
```

Comment:
is this iterating thru all config map keys?

Reply:
yes, will replace all keys (but there is only one), but keep labels/annotations and maybe some other things.